app/src/UI/LegacyMap/helpers/functional/recordset-layers.ts

```diff
@@ -295,20 +295,25 @@ const createOfflineActivitiesLayer = async (
 ) => {
   const geometryList = Object.values(locallyStoredActivities)
     .map((item) => {
-      const parsedData = JSON.parse((item as OfflineActivityRecord).data);
-
-      if (parsedData.geometry && parsedData.geometry[0]) {
-        return {
-          ...parsedData.geometry[0],
-          properties: {
-            short_id: parsedData.short_id
-          }
-        };
+      try {
+        const parsedData = JSON.parse((item as OfflineActivityRecord)?.data);
+        if (parsedData && parsedData.geometry && parsedData.geometry[0]) {
+          return {
+            ...parsedData.geometry[0],
+            properties: {
+              short_id: parsedData.short_id
+            }
+          };
+        }
+
+        return null;
+      } catch (error) {
+        return null;
       }
```

Comment:
```suggestion
      } catch (error) {
        console.error(error);
      }
      return null;
```

can reduce to one return